interfaces/builtin/raw_input.go

```diff
@@ -0,0 +1,77 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/udev"
+)
+
+const rawInputSummary = `allows access to raw input devices`
+
+const rawInputBaseDeclarationSlots = `
+  raw-input:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const rawInputConnectedPlugAppArmor = `
+# Description: Allow reading and writing to raw input devices
+
+/dev/input/* rw,
+
+# Allow reading for supported event reports for all input devices. See
+# https://www.kernel.org/doc/Documentation/input/event-codes.txt
+/sys/devices/**/input[0-9]*/capabilities/* r,
+
+# For using udev
+/run/udev/data/c13:[0-9]* r,
+/run/udev/data/+input:input[0-9]* r,
```

Comment:
Some input devices may actually be from acpi (eg. LNXPWR which is a power button), or from platform (iirc gpios would lead there). Not sure we need to allow reading those right now. Maybe it's better to wait until someone complains.

Reply:
As a matter of facts, I see these messages in the audit log when running the libinput test application (in `--devmode`, that's why they appear as `ALLOWED`:
```
[99572.990328] audit: type=1400 audit(1620722794.063:2261): apparmor="ALLOWED" operation="open" profile="snap.libinput-tool.libinput-debug-events" name="/run/udev/data/+acpi:LNXPWRBN:00" pid=458021 comm="libinput-debug-" requested_mask="r" denied_mask="r" fsuid=0 ouid=0
[99572.990351] audit: type=1400 audit(1620722794.063:2262): apparmor="ALLOWED" operation="open" profile="snap.libinput-tool.libinput-debug-events" name="/run/udev/data/+acpi:LNXSYSTM:00" pid=458021 comm="libinput-debug-" requested_mask="r" denied_mask="r" fsuid=0 ouid=0
[99572.990671] audit: type=1400 audit(1620722794.063:2263): apparmor="ALLOWED" operation="open" profile="snap.libinput-tool.libinput-debug-events" name="/run/udev/data/+acpi:PNP0C0C:00" pid=458021 comm="libinput-debug-" requested_mask="r" denied_mask="r" fsuid=0 ouid=0
[99572.990695] audit: type=1400 audit(1620722794.063:2264): apparmor="ALLOWED" operation="open" profile="snap.libinput-tool.libinput-debug-events" name="/run/udev/data/+acpi:LNXSYBUS:00" pid=458021 comm="libinput-debug-" requested_mask="r" denied_mask="r" fsuid=0 ouid=0
[99572.990720] audit: type=1400 audit(1620722794.063:2265): apparmor="ALLOWED" operation="open" profile="snap.libinput-tool.libinput-debug-events" name="/run/udev/data/+acpi:LNXSYSTM:00" pid=458021 comm="libinput-debug-" requested_mask="r" denied_mask="r" fsuid=0 ouid=0
[99572.991140] audit: type=1400 audit(1620722794.063:2266): apparmor="ALLOWED" operation="open" profile="snap.libinput-tool.libinput-debug-events" name="/run/udev/data/+usb:1-10:1.0" pid=458021 comm="libinput-debug-" requested_mask="r" denied_mask="r" fsuid=0 ouid=0
[99572.991180] audit: type=1400 audit(1620722794.063:2267): apparmor="ALLOWED" operation="open" profile="snap.libinput-tool.libinput-debug-events" name="/run/udev/data/c189:3" pid=458021 comm="libinput-debug-" requested_mask="r" denied_mask="r" fsuid=0 ouid=0
```
I'm now trying to see if I can install the application with strict confinement, to see if these accesses are really needed.

Reply:
c189 is for USB Serial Converters as per https://www.kernel.org/doc/Documentation/admin-guide/devices.txt so I don't think this is relevant for this interface (not sure why libinput is trying to read this but I think this can be safely ignored).

I also agree that for the ACPI related ones as it is not as clear whether these make sense for this interface, perhaps these should be ignored for now.

Reply:
@mardy 

> I'm now trying to see if I can install the application with strict confinement, to see if these accesses are really needed.

what was the outcome of that? I suppose those denials weren't blockers?

As other have mentioned I don't we want to add ACPI devices or c189 here

Reply:
The result is that these accesses are not really needed, the application works fine without them.